### PR TITLE
feat: add longhorn/types

### PR DESCRIPTION
**Pull request by FrankYang0529**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/6744

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/longhorn/longhorn-spdk-engine
 
-go 1.21
+go 1.21.7
```
Comment (innobead) [inline go.mod:3]: Why do we want to use specific Go version here?

**Review by innobead — COMMENTED** [review]
Just one comment, LGTM.

File: go.mod (hunk 1 above)
Comment (FrankYang0529) [reply go.mod]: Removed it. Thanks.

**Review by derekbit — APPROVED** [review]
LGTM

File: Dockerfile.dapper [hunk 2]
```diff
@@ -43,16 +41,16 @@ RUN git clone https://github.com/longhorn/spdk.git ${SPDK_DIR} --recursive && \
     git submodule update --init && \
     ./scripts/pkgdep.sh && \
     if [ ${ARCH} = "amd64" ]; then \
-        ./configure --target-arch=nehalem --disable-tests --disable-unit-tests --disable-examples && \
-        make -j$(nproc) && \
-        make install; \
+    ./configure --target-arch=nehalem --disable-tests --disable-unit-tests --disable-examples && \
+    make -j$(nproc) && \
+    make install; \
```
Comment (derekbit) [inline Dockerfile.dapper:46]: indents?

Comment (FrankYang0529) [reply Dockerfile.dapper]: Reverted it. Thanks.

**Review by innobead — APPROVED** [review]

**Pull request merged**